cumulus/parachains/runtimes/test-utils/src/lib.rs

```diff
@@ -331,14 +338,36 @@ where
 			AllPalletsWithoutSystem::on_initialize(next_block_number);
 
 			let parent_head = HeadData(header.encode());
-			let sproof_builder = RelayStateSproofBuilder {
+			// This ensures the validation in cumulus_pallet_parachain_system passes
+			const NUM_DESCENDANTS: u64 = 2;
+			let authorities = generate_authority_pairs(NUM_DESCENDANTS);
+			let auth_pair = convert_to_authority_weight_pair(&authorities);
+
+			let mut sproof_builder = RelayStateSproofBuilder {
 				para_id: <Runtime>::SelfParaId::get(),
 				included_para_head: parent_head.clone().into(),
 				..Default::default()
 			};
 
+			// Add authorities to the sproof builder
+			use cumulus_primitives_core::relay_chain::well_known_keys;
+			sproof_builder
+				.additional_key_values
+				.push((well_known_keys::AUTHORITIES.to_vec(), auth_pair.clone().encode()));
+			sproof_builder
+				.additional_key_values
+				.push((well_known_keys::NEXT_AUTHORITIES.to_vec(), auth_pair.encode()));
```

Comment:
All this functionality should be part of the `RelayStateSproofBuilder`. Together with all the logic to generate the authority pairs etc. So, people can just directly use it from there.